src/App.js

```diff
@@ -230,58 +191,58 @@ class App extends React.Component {
   }
 
   render() {
+    const selectAlgoOptions = Object.entries(this.state.algos).map(
+      ([algoId, algoDefinition]) => ({
+        value: algoId,
+        label: algoDefinition.name,
+      })
+    );
+    const selectedOption = find(selectAlgoOptions, {
+      value: this.state.selectedAlgoId,
+    });
+
     return (
       <div>
-        <div style={{ width: "100%" }}>
-          <ReactModal
+        <div>
+          <JsonEditor
             isOpen={this.state.showEditor}
-            contentLabel="Minimal Modal Example"
-          >
-            <JSONInput
-              locale={locale}
-              placeholder={this.state.jsonContent}
-              waitAfterKeyPress={2000}
-              theme={"light_mitsuketa_tribute"}
-              colors={{ default: "#888888" }}
-              style={{ body: { fontSize: "16px" } }}
-              onChange={this.onJsonChange}
-            />
-            <button onClick={this.closeEditor}>Cancel</button>
-            <button onClick={this.submitJson}>OK</button>
-          </ReactModal>
+            initialJson={this.state.algos[this.state.selectedAlgoId]}
+            onSubmit={this.submitJson}
+            onCancel={this.closeEditor}
+          />
+        </div>
+        <div style={{ width: "300px", float: "left" }}>
           <Select
-            value={this.state.selectedAlgoOption}
+            value={selectedOption}
             onChange={this.handleAlgoChange}
-            options={filter(this.state.algoOptions, { enabled: true })}
+            options={selectAlgoOptions}
           />
-          <div style={{ width: "300px", float: "left" }}>
-            <Select
-              value={this.state.selectedMetroOption}
-              onChange={this.handleMetroChange}
-              options={filter(metroOptions, { enabled: true })}
-            />
-            <button onClick={this.regenerateData}>Regenerate</button>
-            <button onClick={this.openEditor}>Custom Algo</button>
-
-            <Loader
-              type="Audio"
-              color="#00BFFF"
-              height={100}
-              width={100}
-              timeout={60000} //60 secs
-              visible={this.state.showSpinner}
-            />
-            <RouteCharts
-              hideCharts={this.state.showSpinner}
-              chartData={this.state.chartData}
-            />
-            <button onClick={this.downloadData}>Download</button>
-          </div>
+          <button onClick={this.openEditor}>Add or update...</button>
```

Comment:
I think it might be clearer / less confusing to have separate 'Duplicate'/Edit/Delete buttons.